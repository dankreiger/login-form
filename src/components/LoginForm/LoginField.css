/* form */
/* buttons */
/*** facebook ***/
/*** submit ***/
/* errors */
.LoginField-input, .LoginField-label {
  font-size: 0.81em;
  font-weight: 300; }

/* error message animation */
@keyframes slide-down {
  from {
    transform: translateY(-50%);
    opacity: 0; } }

@media (min-width: 480px) {
  font-size: 0.8em; }

@-webkit-keyframes form_shake {
  0% {
    -webkit-transform: translate(4.5px);
    transform: translate(4.5px); }
  20% {
    -webkit-transform: translate(-4.5px);
    transform: translate(-4.5px); }
  40% {
    -webkit-transform: translate(2px);
    transform: translate(2px); }
  60% {
    -webkit-transform: translate(-2px);
    transform: translate(-2px); }
  80% {
    -webkit-transform: translate(1px);
    transform: translate(1px); }
  100% {
    -webkit-transform: translate(0px);
    transform: translate(0px); } }

.LoginField input[type=checkbox].LoginField-anzeigen {
  margin-right: 0.2rem;
  margin-top: -0.1rem;
  vertical-align: middle; }

.LoginField-input, .LoginField-label {
  /* override browser input autocomplete style */ }
  .LoginField-input:-webkit-autofill, .LoginField-label:-webkit-autofill {
    -webkit-box-shadow: 0 0 0 1000px #FFFFFF inset;
    -webkit-appearance: none; }
  .LoginField-input:focus, .LoginField-label:focus {
    outline: none;
    box-shadow: none;
    border: 1px solid #CCCCCC; }

.LoginField-input {
  height: auto;
  padding: 10px;
  font-size: inherit; }
  .LoginField-input-error {
    border: 1px solid #E3AB57; }
    .LoginField-input-error:focus, .LoginField-input-error:active {
      border: 1px solid #E3AB57; }

.LoginField-label {
  text-transform: uppercase; }

.LoginField-error-message {
  position: absolute;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  height: 23px;
  color: #F4B159;
  font-size: 0.65em;
  background-color: #F7E2AB;
  margin-top: 0.5rem;
  transition: 200ms transform;
  -webkit-animation: slide-down 0.3s backwards ease-in-out;
  animation: slide-down 0.3s backwards ease-in-out; }
  .LoginField-error-message:after {
    bottom: 100%;
    right: 10%;
    border: solid transparent;
    content: " ";
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
    border-color: rgba(247, 226, 171, 0);
    border-bottom-color: #F7E2AB;
    border-width: 7px;
    margin-left: -7px; }
