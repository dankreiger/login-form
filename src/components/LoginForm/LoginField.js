import React, { Component } from 'react';
import { Control, Errors } from 'react-redux-form';

import { FormGroup, InputGroup, Glyphicon, ControlLabel } from 'react-bootstrap';


class LoginField extends Component {
  state = {
    pristine: true,
    valid: null,
    checked: false,
  }

  handleChange = e => {
    this.setState({ pristine: false, valid: this.props.model === '.email' ? e.target.value.length && this.validateEmail(e.target.value) : e.target.value.length });
  }

  handleChecked = e => {
    this.setState({ checked: e.target.checked });
  }

  handleFocus = e => {
    e.target.setSelectionRange(0, e.target.value.length)
  }

  validateEmail = val =>
    this.props.model !== '.email' || !val.length || (val && /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(val))


  render(){
    const { submitFailed, label, glyph, model, checkbox } = this.props, { pristine, valid } = this.state;
    return(
      <div className="LoginField field">
        <FormGroup className="LoginForm-group">
          <ControlLabel className="LoginField-label">{label}</ControlLabel>
          <InputGroup>
            <InputGroup.Addon>
              <Glyphicon glyph={glyph} />
            </InputGroup.Addon>
            <Control.text
              model={model}
              onFocus={this.handleFocus}
              type={model === '.passwort' && !this.state.checked ? 'password' : 'text'}
              validators={{
                validateNotEmpty: val => val && val.length,
                validEmail: val => this.validateEmail(val)
              }}
              validateOn="change"
              onChange={this.handleChange}
              className={`LoginField-input form-control ${!submitFailed && pristine ? '' : valid ? '' : 'LoginField-input-error'}`}
            />
            {checkbox &&
            <InputGroup.Addon>
              <input type="checkbox" className="LoginField-anzeigen" onChange={this.handleChecked}/> Anzeigen
            </InputGroup.Addon>}
          </InputGroup>
          <Errors
            className="LoginField-error-message"
            model={model}
            show={submitFailed || !pristine}
            messages={{
              validateNotEmpty: 'muss ausgefüllt werden',
              validEmail: 'ungültige E-Mail-Adresse',
            }}
          />
        </FormGroup>
      </div>
    )
  }
}


export default LoginField;
