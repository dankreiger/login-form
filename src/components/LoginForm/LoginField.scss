@import '../../variables.scss';
@import './partial_styles/LoginForm-animation.scss';

.LoginField {
  &-anzeigen {
    margin-right: 0.5rem;
  }
  &-input,
  &-label {
    @extend %formText;

    /* override browser input autocomplete style */
    &:-webkit-autofill {
      -webkit-box-shadow: 0 0 0 1000px #FFFFFF inset;
      -webkit-appearance: none;
    }

    &:focus {
      outline: none;
      box-shadow: none;
      border: 1px solid #CCCCCC;
    }
  }

  &-input {
    height: auto;
    padding: 10px;

    &-error {
      border: 1px solid $errorBorderOrange;
      &:focus, &:active {
        border: 1px solid $errorBorderOrange;
      }
    }
  }

  &-label {
    text-transform: uppercase;
  }

  &-error-message {
    position: absolute;
    right: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 50%;
    height: 25px;
    color: $errorColorOrange;
    font-size: 0.65em;
    background-color: $errorBackgroundColorOrange;
    margin-top: 0.8rem;
    margin-bottom: -0.8rem;
    transition: 200ms transform;
    -webkit-animation: slide-down 0.3s backwards ease-in-out;
    animation: slide-down 0.3s backwards ease-in-out;

    &:after {
      bottom: 100%;
      right: 10%;
      border: solid transparent;
      content: " ";
      height: 0;
      width: 0;
      position: absolute;
      pointer-events: none;
      border-color: rgba(247, 226, 171, 0);
      border-bottom-color: $errorBackgroundColorOrange;
      border-width: 7px;
      margin-left: -7px;
    }
  }
}
