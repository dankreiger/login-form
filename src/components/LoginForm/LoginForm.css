/* form */
/* buttons */
/*** facebook ***/
/*** submit ***/
/* errors */
.LoginForm-separator, .LoginForm-terms-and-conditions, .LoginForm-terms-and-conditions-error-message {
  font-size: 0.81em;
  font-weight: 300; }

.LoginForm-submit, .LoginForm-facebook-btn {
  cursor: pointer;
  background-image: none;
  border: 0;
  border-radius: 3px;
  margin: 0 7px 11px 0;
  color: #fff;
  user-select: none;
  transition: all 200ms;
  width: 100%;
  height: 45px;
  font-size: 110%;
  text-shadow: none; }
  .LoginForm-submit:hover, .LoginForm-facebook-btn:hover, .LoginForm-submit:active, .LoginForm-facebook-btn:active, .LoginForm-submit:focus, .LoginForm-facebook-btn:focus, .LoginForm-submit:active:focus, .LoginForm-facebook-btn:active:focus {
    color: #fff;
    outline: 0; }

.LoginForm-separator {
  display: flex;
  justify-content: center;
  text-transform: uppercase; }
  .background.LoginForm-separator {
    position: relative;
    z-index: 1; }
    .background.LoginForm-separator:before {
      border-top: 1px solid #DFDFDF;
      content: "";
      margin: 0 auto;
      position: absolute;
      top: 45%;
      left: 0;
      right: 0;
      bottom: 0;
      width: 95%;
      z-index: -1; }
    .background.LoginForm-separator span {
      background: #FDFDFD;
      padding: 0 15px; }

/* error message animation */
@keyframes slide-down {
  from {
    transform: translateY(-50%);
    opacity: 0; } }

@media (min-width: 480px) {
  font-size: 0.8em; }

@-webkit-keyframes form_shake {
  0% {
    -webkit-transform: translate(4.5px);
    transform: translate(4.5px); }
  20% {
    -webkit-transform: translate(-4.5px);
    transform: translate(-4.5px); }
  40% {
    -webkit-transform: translate(2px);
    transform: translate(2px); }
  60% {
    -webkit-transform: translate(-2px);
    transform: translate(-2px); }
  80% {
    -webkit-transform: translate(1px);
    transform: translate(1px); }
  100% {
    -webkit-transform: translate(0px);
    transform: translate(0px); } }

.LoginForm {
  width: 90%;
  background-color: #FDFDFD;
  padding: 1.2rem 0.8rem;
  border-radius: 10px; }
  @media (min-width: 660px) {
    .LoginForm {
      width: 70%;
      padding-left: 0.9rem;
      padding-right: 0.9rem; } }
  @media (min-width: 760px) {
    .LoginForm {
      width: 60%;
      padding-left: 1rem;
      padding-right: 1rem; } }
  @media (min-width: 860px) {
    .LoginForm {
      width: 50%;
      padding-left: 1.1rem;
      padding-right: 1.1rem; } }
  @media (min-width: 960px) {
    .LoginForm {
      width: 40%;
      padding-left: 1.2rem;
      padding-right: 1.2rem; } }
  .LoginForm-group {
    position: relative;
    margin-bottom: 10px; }
  .LoginForm-submit, .LoginForm-facebook-btn {
    background-color: #739E23;
    box-shadow: 0 2px #618120; }
    .LoginForm-submit:hover, .LoginForm-facebook-btn:hover {
      background-color: #618C23;
      box-shadow: 0 2px #435100; }
    .LoginForm-submit:focus, .LoginForm-facebook-btn:focus {
      outline: 0;
      background-color: #739E23; }
    .LoginForm-submit:active, .LoginForm-facebook-btn:active, .LoginForm-submit:active:focus, .LoginForm-facebook-btn:active:focus {
      text-shadow: none;
      background-color: #618C23;
      box-shadow: 0 0 #717a33;
      top: 4px;
      outline: 0; }
  .LoginForm-facebook-btn {
    background-color: #3C5B95;
    box-shadow: 0 2px #2B0100;
    font-size: 90%;
    position: relative; }
    .LoginForm-facebook-btn i.fb-icon {
      position: absolute;
      left: 5%;
      top: 34%; }
    .LoginForm-facebook-btn:hover {
      background-color: #2B5C94;
      box-shadow: 0 2px #2B4B94; }
    .LoginForm-facebook-btn:focus {
      background-color: #3C5B95; }
    .LoginForm-facebook-btn:active, .LoginForm-facebook-btn:active:focus {
      background-color: #3C5B95;
      box-shadow: 0 1px #2B0100;
      top: 4px; }
  .LoginForm-terms-and-conditions {
    margin-top: 30px;
    margin-bottom: 5px; }
    .LoginForm-terms-and-conditions-error-message {
      color: #F4B159;
      transition: 200ms font-size; }
    .LoginForm-terms-and-conditions input[type="checkbox"]:focus {
      outline: 0; }
  .LoginForm.shakeForm {
    animation: form_shake 0.4s 1 linear;
    -webkit-animation: form_shake 0.4s 1 linear; }
    .LoginForm.shakeForm .LoginField-error-message {
      transform: scale(1.01); }
    .LoginForm.shakeForm .LoginForm-terms-and-conditions-error-message {
      font-size: 0.7em; }
