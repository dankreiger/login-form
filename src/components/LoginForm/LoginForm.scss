%formText {
  font-size: 0.86em;
  font-weight: 300;
}

.LoginForm {
  flex-basis: 75%;
  background-color: #FDFDFD;
  padding: 1.2rem;
  border-radius: 10px;
  @for $i from 0 through 3 {
    @media (min-width: #{660 + (100 * $i)}px) {
      flex-basis: percentage((63 - (10 * $i))/100);
    }
  }

  &-terms-and-conditions {
    @extend %formText;
  }

  .LoginField {
    &-input,
    &-label {
      @extend %formText;

      &:-webkit-autofill {
        -webkit-box-shadow: 0 0 0 1000px white inset;
        -webkit-appearance: none;
      }

      &:focus {
        outline: none;
        box-shadow: none;
        border: 1px solid #ccc;
      }
    }

    &-input {
      height: auto;
      padding: 10px;

      &-error {
        border: 2px solid #E3AB57;
      }
    }

    &-label {
      text-transform: uppercase;
    }

    &-error-message {
      font-size: 0.8rem;
      color: #E3AB57;
      background-color: #F7E2AB;
      margin-top: 0.25rem;
      animation: slide-down 0.3s backwards ease-in-out;
      @keyframes slide-down {
        from {
          transform: translateY(-50%);
          opacity: 0;
        }
      }
    }
  }
}
