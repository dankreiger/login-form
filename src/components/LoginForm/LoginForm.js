import React, { Component } from 'react';
import { Control, Form } from 'react-redux-form';
import { FormGroup, Checkbox, Button } from 'react-bootstrap';
import 'font-awesome/css/font-awesome.min.css';

import LoginField from './LoginField';
import './LoginForm.css';

const fields = [
  {label: 'vorname', glyph: 'user', model: '.vorname'},
  {label: 'nachname', glyph: 'user', model: '.nachname'},
  {label: 'mitgliedsname', glyph: 'user', model: '.mitgliedsname'},
  {label: 'email', glyph: 'user', model: '.email'},
  {label: 'passwort', glyph: 'user', model: '.passwort', checkbox: true},

]

class LoginForm extends Component {
  state = { submitFailed: false, tocAccepted: false }

  handleChangeChk = e => {
    this.setState({tocAccepted: e.target.checked});
  }
  handleClick = e => {
    e.preventDefault();
    let formData = new FormData(document.forms[0]);
    for (var pair of formData.entries()) {
      if(!pair[1]){
        this.setState({submitFailed: true});
        break;
      }
    }
    if(!this.state.tocAccepted) {
      this.setState({submitFailed: true});
    } else {
      this.setState({submitFailed: false});
      document.forms[0].submit();
    }
  }

  render() {
    const {submitFailed, tocAccepted} = this.state;
    return(
      <Form
        className="LoginForm"
        model="user"
        onSubmit={v => alert(`Formular wurde erfolgreich übermittelt\n: ${JSON.stringify(v, null, 4)}`)}>
        <Button className="LoginForm-facebook-btn" type="button" >
          <i className="fa fa-facebook"></i>
          Über Facebook registrieren
        </Button>
        <div className="LoginForm-separator background"><span>oder ohne facebook</span></div>
        {fields.map(field =>
          <LoginField
            key={field.label}
            label={field.label}
            glyph={field.glyph}
            model={field.model}
            checkbox={field.checkbox}
            submitFailed={submitFailed}
          />
        )}
        <FormGroup>
          <Checkbox className="LoginForm-terms-and-conditions" name="terms-and-conditions" onChange={this.handleChangeChk}>
            Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.
          </Checkbox>
          {submitFailed && !tocAccepted &&
            <div className="LoginForm-terms-and-conditions-error-message">
              * Sie müssen die Nutzungsbedingungen akzeptieren
            </div>
          }
        </FormGroup>
        <Button bsStyle='success' className="LoginForm-submit" type="button" onClick={this.handleClick}>
          Submit
        </Button>
      </Form>
    )
  }
}





export default LoginForm;
