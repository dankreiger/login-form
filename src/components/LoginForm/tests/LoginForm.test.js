import React from 'react';
import {shallow} from 'enzyme';
import LoginForm from '../LoginForm';

const loginform = shallow(<LoginForm />);

it('renders without crashing', () => {
  expect(loginform).toMatchSnapshot();
});
