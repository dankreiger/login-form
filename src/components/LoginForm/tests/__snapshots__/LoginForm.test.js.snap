// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <t
          appId="9999999999999999"
          authType=""
          autoLoad={true}
          callback={[Function]}
          cookie={false}
          disableMobileRedirect={false}
          fields="name"
          isMobile={false}
          language="en_US"
          onFailure={null}
          redirectUri="http://localhost/"
          render={[Function]}
          responseType="code"
          returnScopes={false}
          scope="public_profile,email"
          state="facebookdirect"
          version="2.3"
          xfbml={false}
/>,
        <div
          className="LoginForm-separator background"
>
          <span>
                    oder ohne facebook
          </span>
</div>,
        Array [
          <LoginField
            checkbox={undefined}
            glyph="user"
            label="vorname"
            model=".vorname"
            submitFailed={false}
/>,
          <LoginField
            checkbox={undefined}
            glyph="user"
            label="nachname"
            model=".nachname"
            submitFailed={false}
/>,
          <LoginField
            checkbox={undefined}
            glyph="user"
            label="mitgliedsname"
            model=".mitgliedsname"
            submitFailed={false}
/>,
          <LoginField
            checkbox={undefined}
            glyph="user"
            label="email"
            model=".email"
            submitFailed={false}
/>,
          <LoginField
            checkbox={true}
            glyph="user"
            label="passwort"
            model=".passwort"
            submitFailed={false}
/>,
        ],
        <FormGroup
          bsClass="form-group"
>
          <Checkbox
                    bsClass="checkbox"
                    className="LoginForm-terms-and-conditions"
                    disabled={false}
                    inline={false}
                    name="terms-and-conditions"
                    onChange={[Function]}
                    title=""
          >
                    Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.
          </Checkbox>
</FormGroup>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="LoginForm-submit"
          disabled={false}
          onClick={[Function]}
          type="button"
>
          Submit
</Button>,
      ],
      "className": "LoginForm",
      "model": "user",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appId": "9999999999999999",
          "authType": "",
          "autoLoad": true,
          "callback": [Function],
          "cookie": false,
          "disableMobileRedirect": false,
          "fields": "name",
          "isMobile": false,
          "language": "en_US",
          "onFailure": null,
          "redirectUri": "http://localhost/",
          "render": [Function],
          "responseType": "code",
          "returnScopes": false,
          "scope": "public_profile,email",
          "state": "facebookdirect",
          "version": "2.3",
          "xfbml": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            oder ohne facebook
</span>,
          "className": "LoginForm-separator background",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "oder ohne facebook",
          },
          "ref": null,
          "rendered": "oder ohne facebook",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "vorname",
        "nodeType": "class",
        "props": Object {
          "checkbox": undefined,
          "glyph": "user",
          "label": "vorname",
          "model": ".vorname",
          "submitFailed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "nachname",
        "nodeType": "class",
        "props": Object {
          "checkbox": undefined,
          "glyph": "user",
          "label": "nachname",
          "model": ".nachname",
          "submitFailed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "mitgliedsname",
        "nodeType": "class",
        "props": Object {
          "checkbox": undefined,
          "glyph": "user",
          "label": "mitgliedsname",
          "model": ".mitgliedsname",
          "submitFailed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "email",
        "nodeType": "class",
        "props": Object {
          "checkbox": undefined,
          "glyph": "user",
          "label": "email",
          "model": ".email",
          "submitFailed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "passwort",
        "nodeType": "class",
        "props": Object {
          "checkbox": true,
          "glyph": "user",
          "label": "passwort",
          "model": ".passwort",
          "submitFailed": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": Array [
            <Checkbox
              bsClass="checkbox"
              className="LoginForm-terms-and-conditions"
              disabled={false}
              inline={false}
              name="terms-and-conditions"
              onChange={[Function]}
              title=""
>
              Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.
</Checkbox>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "checkbox",
              "children": "Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.",
              "className": "LoginForm-terms-and-conditions",
              "disabled": false,
              "inline": false,
              "name": "terms-and-conditions",
              "onChange": [Function],
              "title": "",
            },
            "ref": null,
            "rendered": "Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.",
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Submit",
          "className": "LoginForm-submit",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Submit",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <t
            appId="9999999999999999"
            authType=""
            autoLoad={true}
            callback={[Function]}
            cookie={false}
            disableMobileRedirect={false}
            fields="name"
            isMobile={false}
            language="en_US"
            onFailure={null}
            redirectUri="http://localhost/"
            render={[Function]}
            responseType="code"
            returnScopes={false}
            scope="public_profile,email"
            state="facebookdirect"
            version="2.3"
            xfbml={false}
/>,
          <div
            className="LoginForm-separator background"
>
            <span>
                        oder ohne facebook
            </span>
</div>,
          Array [
            <LoginField
              checkbox={undefined}
              glyph="user"
              label="vorname"
              model=".vorname"
              submitFailed={false}
/>,
            <LoginField
              checkbox={undefined}
              glyph="user"
              label="nachname"
              model=".nachname"
              submitFailed={false}
/>,
            <LoginField
              checkbox={undefined}
              glyph="user"
              label="mitgliedsname"
              model=".mitgliedsname"
              submitFailed={false}
/>,
            <LoginField
              checkbox={undefined}
              glyph="user"
              label="email"
              model=".email"
              submitFailed={false}
/>,
            <LoginField
              checkbox={true}
              glyph="user"
              label="passwort"
              model=".passwort"
              submitFailed={false}
/>,
          ],
          <FormGroup
            bsClass="form-group"
>
            <Checkbox
                        bsClass="checkbox"
                        className="LoginForm-terms-and-conditions"
                        disabled={false}
                        inline={false}
                        name="terms-and-conditions"
                        onChange={[Function]}
                        title=""
            >
                        Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.
            </Checkbox>
</FormGroup>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="LoginForm-submit"
            disabled={false}
            onClick={[Function]}
            type="button"
>
            Submit
</Button>,
        ],
        "className": "LoginForm",
        "model": "user",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appId": "9999999999999999",
            "authType": "",
            "autoLoad": true,
            "callback": [Function],
            "cookie": false,
            "disableMobileRedirect": false,
            "fields": "name",
            "isMobile": false,
            "language": "en_US",
            "onFailure": null,
            "redirectUri": "http://localhost/",
            "render": [Function],
            "responseType": "code",
            "returnScopes": false,
            "scope": "public_profile,email",
            "state": "facebookdirect",
            "version": "2.3",
            "xfbml": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              oder ohne facebook
</span>,
            "className": "LoginForm-separator background",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "oder ohne facebook",
            },
            "ref": null,
            "rendered": "oder ohne facebook",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "vorname",
          "nodeType": "class",
          "props": Object {
            "checkbox": undefined,
            "glyph": "user",
            "label": "vorname",
            "model": ".vorname",
            "submitFailed": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "nachname",
          "nodeType": "class",
          "props": Object {
            "checkbox": undefined,
            "glyph": "user",
            "label": "nachname",
            "model": ".nachname",
            "submitFailed": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "mitgliedsname",
          "nodeType": "class",
          "props": Object {
            "checkbox": undefined,
            "glyph": "user",
            "label": "mitgliedsname",
            "model": ".mitgliedsname",
            "submitFailed": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "email",
          "nodeType": "class",
          "props": Object {
            "checkbox": undefined,
            "glyph": "user",
            "label": "email",
            "model": ".email",
            "submitFailed": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "passwort",
          "nodeType": "class",
          "props": Object {
            "checkbox": true,
            "glyph": "user",
            "label": "passwort",
            "model": ".passwort",
            "submitFailed": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <Checkbox
                bsClass="checkbox"
                className="LoginForm-terms-and-conditions"
                disabled={false}
                inline={false}
                name="terms-and-conditions"
                onChange={[Function]}
                title=""
>
                Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.
</Checkbox>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "checkbox",
                "children": "Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.",
                "className": "LoginForm-terms-and-conditions",
                "disabled": false,
                "inline": false,
                "name": "terms-and-conditions",
                "onChange": [Function],
                "title": "",
              },
              "ref": null,
              "rendered": "Ich willige in die Verarbeitung und Nutzung meiner Daten gemäß der Datenschutzerklärung ein.",
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Submit",
            "className": "LoginForm-submit",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
